src/devrules/core/git_service.py

```
@@ -448,3 +417,100 @@ def get_current_repo_name() -> str:
+            ["git", "push", "-u", "origin", branch],
+            check=True,
+        )
+    except subprocess.CalledProcessError as e:
+        return False, str(e)
+
+
+def check_not_pushed_changes() -> bool:
+    """Check if there are changes that have not been pushed."""
+    try:
+        result = subprocess.run(
+            ["git", "status"],
```

Ruff (0.14.13)

Starting a process with a partial executable path

(S607)